superfluous_disable_command reports violations on words in comment that are not rules

### New Issue Checklist

- [ 0.30.1 ] Updated SwiftLint to the latest version
- [ x ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Swiftlint reports violations for extraneous words in the same comment that follow the swiftlint rule. Many of our team have written disables that look like the one below in the example. Swiftlint should not report a violation for words that it finds that are not known rules.

The example below also reports the correct violation 79 times when I would expect only one. In combination this results in excessive reporting in some situations with the 7 lines of code below reporting 869 violations which is 11 full repeats.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files at paths 
Linting 'File.swift' (1/1)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'no_indentation_spaces' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'no_indentation_spaces' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'no_indentation_spaces' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
...
[deleted 813 lines]
...
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'no_indentation_spaces' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'no_indentation_spaces' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'no_indentation_spaces' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:1:73: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'as' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'this' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'doesn't' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'expected' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
File.swift:6:42: warning: Superfluous Disable Command Violation: SwiftLint rule 'work' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
Done linting! Found 869 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
	0.30.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
	Built from source
* Paste your configuration file:
	base configuration

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
	No
* Which Xcode version are you using (check `xcode-select -p`)?
	10.1 (10B61)
* Do you have a sample that shows the issue?
	Yes

```swift
// swiftlint:disable no_indentation_spaces this doesn't work as expected
func loggingFunctionTest(_ message: @autoclosure () -> String,
                         file: StaticString = #file,
                         function: StaticString = #function,
                         line: UInt = #line) {
// swiftlint:enable no_indentation_spaces
}
```

Full Output:
[output.txt](https://github.com/realm/SwiftLint/files/2821795/output.txt)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -60,6 +60,12 @@
   [Igor-Palaguta](https://github.com/Igor-Palaguta)
   [#2544](https://github.com/realm/SwiftLint/issues/2544)
 
+* Fix excessive `superfluous_disable_command` violations being reported when
+  using an invalid rule identifier in a disable command.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2623](https://github.com/realm/SwiftLint/issues/2623)
+
+  
 ## 0.30.1: Localized Stain Remover
 
 #### Breaking

--- a/Source/SwiftLintFramework/Models/Command.swift
+++ b/Source/SwiftLintFramework/Models/Command.swift
@@ -82,7 +82,8 @@ public struct Command: Equatable {
         let ruleTexts = scanner.string.bridge().substring(
             from: scanner.scanLocation + 1
         ).components(separatedBy: .whitespaces).filter {
-            !$0.trimmingCharacters(in: .whitespaces).isEmpty
+            let component = $0.trimmingCharacters(in: .whitespaces)
+            return !component.isEmpty && component != "*/"
         }
         ruleIdentifiers = Set(ruleTexts.map(RuleIdentifier.init(_:)))
 

--- a/Source/SwiftLintFramework/Models/Command.swift
+++ b/Source/SwiftLintFramework/Models/Command.swift
@@ -81,7 +81,9 @@ public struct Command: Equatable {
 
         let ruleTexts = scanner.string.bridge().substring(
             from: scanner.scanLocation + 1
-        ).components(separatedBy: .whitespaces)
+        ).components(separatedBy: .whitespaces).filter {
+            !$0.trimmingCharacters(in: .whitespaces).isEmpty
+        }
         ruleIdentifiers = Set(ruleTexts.map(RuleIdentifier.init(_:)))
 
         // Modifier

--- a/Source/SwiftLintFramework/Models/Linter.swift
+++ b/Source/SwiftLintFramework/Models/Linter.swift
@@ -83,10 +83,6 @@ private extension Rule {
             allViolations: violations
         )
 
-        let undefinedSuperfluousCommandViolations = self.undefinedSuperfluousCommandViolations(
-            regions: regions, configuration: configuration,
-            superfluousDisableCommandRule: superfluousDisableCommandRule)
-
         let enabledViolations: [StyleViolation]
         if file.contents.hasPrefix("#!") { // if a violation happens on the same line as a shebang, ignore it
             enabledViolations = enabledViolationsAndRegions.compactMap { violation, _ in
@@ -101,37 +97,9 @@ private extension Rule {
             return identifiers.map { ($0, ruleID) }
         }
 
-        return LintResult(
-            violations: enabledViolations + superfluousDisableCommandViolations + undefinedSuperfluousCommandViolations,
-            ruleTime: ruleTime,
-            deprecatedToValidIDPairs: deprecatedToValidIDPairs)
-    }
-
-    private func undefinedSuperfluousCommandViolations(regions: [Region],
-                                                       configuration: Configuration,
-                                                       superfluousDisableCommandRule: SuperfluousDisableCommandRule?
-        ) -> [StyleViolation] {
-        guard !regions.isEmpty, let superfluousDisableCommandRule = superfluousDisableCommandRule else {
-            return []
-        }
-        let allCustomIdentifiers = configuration.customRuleIdentifiers.map { RuleIdentifier($0) }
-        let allRuleIdentifiers = masterRuleList.allValidIdentifiers().map { RuleIdentifier($0) }
-        let allValidIdentifiers = Set(allCustomIdentifiers + allRuleIdentifiers)
-        let regions = regions.filter {
-            !$0.disabledRuleIdentifiers.contains(.all) &&
-                $0.disabledRuleIdentifiers.isDisjoint(with: allValidIdentifiers)
-        }
-
-        return regions.flatMap { region in
-            region.disabledRuleIdentifiers.filter({ !allValidIdentifiers.contains($0) }).map { id in
-                return StyleViolation(
-                    ruleDescription: type(of: superfluousDisableCommandRule).description,
-                    severity: superfluousDisableCommandRule.configuration.severity,
-                    location: region.start,
-                    reason: superfluousDisableCommandRule.reason(for: id.stringRepresentation)
-                )
-            }
-        }
+        return LintResult(violations: enabledViolations + superfluousDisableCommandViolations,
+                          ruleTime: ruleTime,
+                          deprecatedToValidIDPairs: deprecatedToValidIDPairs)
     }
 }
 
@@ -168,7 +136,11 @@ public struct Linter {
                     superfluousDisableCommandRule: superfluousDisableCommandRule,
                     compilerArguments: self.compilerArguments)
         }
-        let violations = validationResults.flatMap { $0.violations }
+        let undefinedSuperfluousCommandViolations = self.undefinedSuperfluousCommandViolations(
+            regions: regions, configuration: configuration,
+            superfluousDisableCommandRule: superfluousDisableCommandRule)
+
+        let violations = validationResults.flatMap { $0.violations } + undefinedSuperfluousCommandViolations
         let ruleTimes = validationResults.compactMap { $0.ruleTime }
         var deprecatedToValidIdentifier = [String: String]()
         for (key, value) in validationResults.flatMap({ $0.deprecatedToValidIDPairs }) {
@@ -245,6 +217,29 @@ public struct Linter {
                                                  indentWidth: indentWidth)
         if let formattedContents = formattedContents {
             file.write(formattedContents)
+        }
+    }
+
+    private func undefinedSuperfluousCommandViolations(regions: [Region],
+                                                       configuration: Configuration,
+                                                       superfluousDisableCommandRule: SuperfluousDisableCommandRule?
+        ) -> [StyleViolation] {
+        guard !regions.isEmpty, let superfluousDisableCommandRule = superfluousDisableCommandRule else {
+            return []
+        }
+        let allCustomIdentifiers = configuration.customRuleIdentifiers.map { RuleIdentifier($0) }
+        let allRuleIdentifiers = masterRuleList.allValidIdentifiers().map { RuleIdentifier($0) }
+        let allValidIdentifiers = Set(allCustomIdentifiers + allRuleIdentifiers + [.all])
+
+        return regions.flatMap { region in
+            region.disabledRuleIdentifiers.filter({ !allValidIdentifiers.contains($0) }).map { id in
+                return StyleViolation(
+                    ruleDescription: type(of: superfluousDisableCommandRule).description,
+                    severity: superfluousDisableCommandRule.configuration.severity,
+                    location: region.start,
+                    reason: superfluousDisableCommandRule.reason(forNonExistentRule: id.stringRepresentation)
+                )
+            }
         }
     }
 }

--- a/Source/SwiftLintFramework/Rules/Lint/SuperfluousDisableCommandRule.swift
+++ b/Source/SwiftLintFramework/Rules/Lint/SuperfluousDisableCommandRule.swift
@@ -26,4 +26,8 @@ public struct SuperfluousDisableCommandRule: ConfigurationProviderRule {
         return "SwiftLint rule '\(rule)' did not trigger a violation " +
         "in the disabled region. Please remove the disable command."
     }
+
+    public func reason(forNonExistentRule rule: String) -> String {
+        return "'\(rule)' is not a valid SwiftLint rule. Please remove it from the disable command."
+    }
 }

--- a/Tests/SwiftLintFrameworkTests/CommandTests.swift
+++ b/Tests/SwiftLintFrameworkTests/CommandTests.swift
@@ -260,6 +260,8 @@ class CommandTests: XCTestCase {
             violations("print(123)\n// swiftlint:disable:previous nesting_foo\n")[0].reason,
             "'nesting_foo' is not a valid SwiftLint rule. Please remove it from the disable command."
         )
+
+        XCTAssertEqual(violations("/* swiftlint:disable nesting */\n").count, 1)
     }
 
     func testSuperfluousDisableCommandsDisabled() {

--- a/Tests/SwiftLintFrameworkTests/CommandTests.swift
+++ b/Tests/SwiftLintFrameworkTests/CommandTests.swift
@@ -240,6 +240,21 @@ class CommandTests: XCTestCase {
             violations("print(123)\n// swiftlint:disable:previous nesting_foo\n")[0].ruleDescription.identifier,
             "superfluous_disable_command"
         )
+
+        let multipleViolations = violations("// swiftlint:disable nesting this is a comment\n")
+        XCTAssertEqual(multipleViolations.count, 5)
+        XCTAssertTrue(multipleViolations.allSatisfy { $0.ruleDescription.identifier == "superfluous_disable_command" })
+
+        let onlyNonExistentRulesViolations = violations("// swiftlint:disable this is a comment\n")
+        XCTAssertEqual(onlyNonExistentRulesViolations.count, 4)
+        XCTAssertTrue(onlyNonExistentRulesViolations.allSatisfy {
+            $0.ruleDescription.identifier == "superfluous_disable_command"
+        })
+
+        XCTAssertEqual(
+            violations("print(123)\n// swiftlint:disable:previous nesting_foo\n")[0].reason,
+            "'nesting_foo' is not a valid SwiftLint rule. Please remove it from the disable command."
+        )
     }
 
     func testSuperfluousDisableCommandsDisabled() {

--- a/Tests/SwiftLintFrameworkTests/CommandTests.swift
+++ b/Tests/SwiftLintFrameworkTests/CommandTests.swift
@@ -241,6 +241,11 @@ class CommandTests: XCTestCase {
             "superfluous_disable_command"
         )
 
+        XCTAssertEqual(
+            violations("print(123)\n// swiftlint:disable:previous nesting_foo \n").count,
+            1
+        )
+
         let multipleViolations = violations("// swiftlint:disable nesting this is a comment\n")
         XCTAssertEqual(multipleViolations.count, 5)
         XCTAssertTrue(multipleViolations.allSatisfy { $0.ruleDescription.identifier == "superfluous_disable_command" })

